### PR TITLE
duplicated header when break-after: avoid; active

### PR DESCRIPTION
**Pull request by ahus1**
I tried a `break-after: avoid;` for headings, and this turned out to give duplicated headers at end and start of page. [See page 9 of the PDF for an example.](https://github.com/ahus1/asciidoctor-pdf.js/blob/215b94664a5b231e879d38cf1b7f67c8252dce7d/examples/document/basic-example.pdf).

image

Might be an issue in pagedjs, or a mis-configuration of pagedjs.

### CODE REVIEW

**Comment by Mogztter** [comment]
I did reproduce this issue, see https://gitlab.pagedmedia.org/tools/pagedjs/issues/136

**Comment by ahus1** [comment]
@Mogztter - thanks for pointing me at the other ticket. I've followed the links in that ticket and added the (hacky) workaround mentioned in the ticket. This fixes the issue. 

Ready for merge feature wise. Not sure how to test this in a unit test. Hints for refactoring welcome.

File: lib/document/templates.js [hunk 1]
```diff
@@ -242,9 +242,33 @@ ${pagedContent}
 window.AsciidoctorPDF = window.AsciidoctorPDF || {}
 window.AsciidoctorPDF.status = 'rendering'
 class PagedReadyHandler extends Paged.Handler {
-    afterRendered(pages) {
-        window.AsciidoctorPDF.status = 'ready'
+  afterRendered(pages) {
```
Comment (Mogztter) [inline lib/document/templates.js]: I think it should be moved to a JavaScript file.

**Comment by Mogztter** [comment]
> Not sure how to test this in a unit test.

In a unit test I think it's impossible.
At some point we might introduce a visual comparison tool to create a diff between two PDF: https://github.com/asciidoctor/asciidoctor-pdf/blob/8d1d0d9bd2f63b5bf2fdad6ac4eda9e1bd749f89/spec/spec_helper.rb#L550-L584

**Comment by ahus1** [comment]
On hold waiting for #122, as both change the same (now externalized) file.

**Comment by ahus1** [comment]
Rebased, workaround now part of `paged-rendering.js`. 

@Mogztter - ready for review and merge. Please feel free to add finishing touches where I've missed them (that's always appreciated).

File: lib/document/paged-rendering.js [hunk 2]
```diff
@@ -5,7 +5,33 @@ class PagedReadyHandler extends Paged.Handler {
   afterRendered (pages) {
     window.AsciidoctorPDF.status = 'ready'
   }
+
+  // BEGIN: workaround to prevent duplicated content at end/beginning of page
+  // https://gitlab.pagedmedia.org/tools/pagedjs/issues/126
```
Comment (Mogztter) [inline lib/document/paged-rendering.js]: @ahus1 Did you try the other alternative (ie. remove newlines when generating the HTML code)?

Comment (ahus1) [reply lib/document/paged-rendering.js]: @Mogztter - no, I didn't try fiddling with newlines. This sounded to error prone to me: removing all newlines would have been too much considering pre-formatted content. Removing "just the non-necessary" newlines sounded hard. Not talking about people (like me) using pass-through content to inject some JavaScript here and there...

Comment (Mogztter) [reply lib/document/paged-rendering.js]: You're probably right, hopefully this workaround won't be needed in the future :wink:

**Pull request merged**